apps/condo/domains/resident/schema/DiscoverServiceConsumersService.test.js

```diff
@@ -3,275 +3,990 @@
+                expect(receipts1).toHaveLength(6)
+                expect([...registeredReceipts1, ...registeredReceipts1a, ...registeredReceipts1b, ...registeredReceipts1c, ...registeredReceipts2a]).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts1[0].id }),
+                    expect.objectContaining({ id: receipts1[1].id }),
+                    expect.objectContaining({ id: receipts1[2].id }),
+                    expect.objectContaining({ id: receipts1[3].id }),
+                    expect.objectContaining({ id: receipts1[4].id }),
+                    expect.objectContaining({ id: receipts1[5].id }),
+                ]))
+
+                const receipts2 = await ResidentBillingReceipt.getAll(residentClient2, {})
+                expect(receipts2).toHaveLength(4)
+                expect([...registeredReceipts1b, ...registeredReceipts1c, ...registeredReceipts2, ...registeredReceipts2a]).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts2[0].id }),
+                    expect.objectContaining({ id: receipts2[1].id }),
+                    expect.objectContaining({ id: receipts2[2].id }),
+                    expect.objectContaining({ id: receipts2[3].id }),
+                ]))
+
+                const receipts3 = await ResidentBillingReceipt.getAll(residentClient3, {})
+                expect(receipts3).toHaveLength(6)
+                expect([...registeredReceipts1, ...registeredReceipts1a, ...registeredReceipts1b, ...registeredReceipts1c, ...registeredReceipts2a]).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts3[0].id }),
+                    expect.objectContaining({ id: receipts3[1].id }),
+                    expect.objectContaining({ id: receipts3[2].id }),
+                    expect.objectContaining({ id: receipts3[3].id }),
+                    expect.objectContaining({ id: receipts3[4].id }),
+                    expect.objectContaining({ id: receipts3[5].id }),
+                ]))
+
+                const receipts4 = await ResidentBillingReceipt.getAll(residentClient4, {})
+                expect(receipts4).toHaveLength(4)
+                expect([...registeredReceipts1b, ...registeredReceipts1c, ...registeredReceipts2, ...registeredReceipts2a]).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts4[0].id }),
+                    expect.objectContaining({ id: receipts4[1].id }),
+                    expect.objectContaining({ id: receipts4[2].id }),
+                    expect.objectContaining({ id: receipts4[3].id }),
+                ]))
+            }, { delay: 500 })
+
+            // 5th resident registered to updated address
+            const unitType5 = FLAT_UNIT_TYPE
+            const unitName5 = faker.lorem.word()
+            await registerResidentByTestClient(
```

Comment:
Add case when there's receipts already:
1. Org uploaded big registry for 2 addresses (2 and 2a)
2. They created 2 property
3. They realised their mistake, changed address from 2 to 2a
4. Resident registered and saw receipts from 2a correctly

Reply:
Done